mirrord/layer/src/outgoing/tcp.rs

```diff
@@ -381,22 +381,28 @@ impl TcpOutgoingHandler {
             DaemonTcpOutgoing::Read(read) => {
                 // (agent) read something from remote, so we write it to the user.
                 trace!("Read -> read {:?}", read);
-                let DaemonRead {
-                    connection_id,
-                    bytes,
-                } = read?;
-
-                let sender = self
-                    .mirrors
-                    .get_mut(&connection_id)
-                    .ok_or(LayerError::NoConnectionId(connection_id))?;
-
-                sender.send(bytes).await.unwrap_or_else(|_| {
-                    warn!(
+
+                match read {
+                    Ok(DaemonRead {
+                        connection_id,
+                        bytes,
+                    }) => {
+                        let sender = self
+                            .mirrors
+                            .get_mut(&connection_id)
+                            .ok_or(LayerError::NoConnectionId(connection_id))?;
+
+                        sender.send(bytes).await.unwrap_or_else(|_| {
+                            warn!(
                         "Got new data from agent after application closed socket. connection_id: \
                     connection_id: {connection_id}"
                     );
-                });
+                        });
+                    }
```

Comment:
Add a comment describing the best solution:
We should either:
1. add `DaemonTcpOutgoing::Error(errno)` and then pass it to the `interceptor_task` to break.
2. change `DaemonTcpOutgoing::Close` to `Close(Option<errno>)` and pass it to the `interceptr_task` to know when to shutdown and when to break loop (sounds uglier though)

Reply:
and that we don't do that yet because that'd require breaking the protocol, so we're waiting for #1353